pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -1087,6 +1084,22 @@ private void attachTimeBoundary(String rawTableName, BrokerRequest brokerRequest
     }
   }
 
+  /**
+   * The time boundary filter string splits queries to offline and realtime pinot tables based on the current high
+   * watermark of available segments in the offline table. By default, Pinot serves all segments from offline up to
+   * the latest segment (exclusive), and everything else from realtime. However, one can configure Pinot to serve
+   * the latest offline segment inclusively via config.
+   */
+  private String getTimeBoundaryFilter(String offlineTableName, String timeValue, boolean isOfflineRequest) {
+    boolean shouldServeOfflineSegmentsImmediately = Optional.ofNullable(_tableCache.getTableConfig(offlineTableName).getQueryConfig())
+        .map(QueryConfig::getServeOfflineSegmentsImmediately)
+        .orElse(false);
+    String open = isOfflineRequest || !shouldServeOfflineSegmentsImmediately ? "(" : "[";
```

Comment:
Perhaps a little more self documenting:

```
boolean openRangeInclusive = !isOfflineRequest || shouldServeOfflineSegments;
String open = openRangeInclusive ? INCLUSIVE_OPEN_RANGE : EXCLUSIVE_OPEN_RANGE;
String range = isOfflineRequest ? ALL_VALUES_BEFORE + timeValue : timeValue + ALL_VALUES_AFTER;
boolean closeRangeInclusive = isOfflineRequest && !shouldServeOfflineSegmentsImmediately;
```

Etc

As rightt now this is a little confusing because I assume that `]` is inclusive and `)` is exclusive, but `)` seems to be inclusive when its offline and not configured to serve immediately, which seems counter intuitive.

Reply:
Refactored similar to the above.